src/packagedcode/freebsd.py

```diff
@@ -148,20 +181,21 @@ def license_mapper(package_data, package):
     if not licenses:
         return
 
+    declared_license = OrderedDict()
```

Comment:
I think that we should stay true to the original data structure:
This should be something such as:
`{ 'licenselogic': 'dual', 'licenses': [a list of license strings, ... ]}`

This was we can do the detection much better... e.g. we should NOT try to craft an a pseudo expression in the `declared_license`